### DISCUSSION TITLE
Add functions to check if private key with subkeys is encrypted

### DISCUSSION THREAD

**Discussion by IvanPizhenko** [openpgpjs/openpgpjs#1488] [Ideas]
I suggest to add following functions to the class `PrivateKey`:
- `isFullyEncrypted(): boolean` - returns `true` if all private key and subkey packets are encrypted, otherwise returns `false`.
- `isFullyDecrypted(): boolean` - returns `true` if all private key and subkey packets are decrypted, otherwise returns `false`.
- `isPacketDecrypted(keyID: KeyID): boolean` - returns `true` if private key or subkey with given ID is decrypted, otherwise returns false. Throws exception, if `keyID` is not provided or packet with given `keyID` doesn't exist.

I can try to post PR with these functions.

@larabr Do you support this idea?

**Comment by twiss** [comment]
Hello :wave:

Partially decrypting a key is not really supported, especially now that we have `openpgp.decryptKey({ privateKey, passphrase })` rather than `privateKey.decrypt(passphrases, keyId = null)` like we used to. Even when we had that, it wasn't very well supported. What's your use case for this?
